// beekeeper/dist/model/file.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toFile = exports.FileStatus = exports.FileUsageType = void 0;
const identifier_1 = require("./identifier");
const timestamp_1 = require("./timestamp");
var FileUsageType;
(function (FileUsageType) {
    FileUsageType["ARTIFACT"] = "artifact";
    FileUsageType["ATTACHMENT_FILE"] = "attachment_file";
    FileUsageType["ATTACHMENT_IMAGE"] = "attachment_image";
    FileUsageType["ATTACHMENT_VIDEO"] = "attachment_video";
    FileUsageType["AVATAR"] = "avatar";
    FileUsageType["CHAT_GROUP_IMAGE"] = "chat_group_image";
    FileUsageType["NAVIGATION_EXTENSION_FILE"] = "navigation_extension_file";
    FileUsageType["VOICE_RECORDING"] = "voice_recording";
})(FileUsageType = exports.FileUsageType || (exports.FileUsageType = {}));
var FileStatus;
(function (FileStatus) {
    FileStatus["READY"] = "ready";
    FileStatus["PENDING"] = "pending";
    FileStatus["ERROR"] = "error";
})(FileStatus = exports.FileStatus || (exports.FileStatus = {}));
function toFile(data) {
    if (typeof data !== 'object') {
        throw new Error('toFile() only support plain objects');
    }
    // This assumes that `data` is actually a valid API response.
    return {
        ...data,
        id: new identifier_1.NumericIdentifier(data.id),
        created: timestamp_1.parseTimestamp(data.created),
        updated: timestamp_1.parseTimestamp(data.updated),
        userid: new identifier_1.UUIDIdentifier(data.userid),
    };
}
exports.toFile = toFile;
