// beekeeper/dist/model/message.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toMessage = exports.messageFactory = exports.MessageTypes = void 0;
const identifier_1 = require("./identifier");
/** Enum for the different types of message types */
var MessageTypes;
(function (MessageTypes) {
    MessageTypes["REGULAR"] = "regular";
})(MessageTypes = exports.MessageTypes || (exports.MessageTypes = {}));
function messageFactory(sender, conversation, messageData) {
    const msg = Object.assign({}, messageData, {
        message_type: MessageTypes.REGULAR,
        user_id: sender.id,
        profile: sender.name,
        conversation_id: conversation.id,
    });
    return msg;
}
exports.messageFactory = messageFactory;
// tslint:disable-next-line no-any
function toMessage(data) {
    const uuid = data.uuid ? new identifier_1.UUIDIdentifier(data.uuid) : undefined;
    return Object.assign({}, data, {
        uuid,
        profile: new identifier_1.UsernameIdentifier(data.profile),
        conversation_id: new identifier_1.NumericIdentifier(data.conversation_id),
        user_id: new identifier_1.UUIDIdentifier(data.user_id),
    });
}
exports.toMessage = toMessage;
