// beekeeper/dist/sdk/resource/error.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ResourceError extends Error {
    constructor(status, text, error) {
        const message = error ? error : `${status}: ${text}`;
        super(message);
        this.status = status;
        this.text = text;
        this.error = error;
    }
    static async fromResponse(response) {
        let error;
        if (response.status > 0) {
            error = await ResourceError.getErrorBody(response);
        }
        return new ResourceError(response.status, response.statusText, error);
    }
    static async getErrorBody(response) {
        const responseBodyText = await response.text();
        if (!responseBodyText) {
            return undefined;
        }
        try {
            const responseBody = JSON.parse(responseBodyText);
            if (responseBody.error) {
                return responseBody.error;
            }
            return responseBodyText;
        }
        catch (e) {
            return responseBodyText;
        }
    }
}
exports.default = ResourceError;
