// beekeeper/dist/sdk/resource/post.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostResourceImpl = exports.PostsFilter = void 0;
const model_1 = require("../../model");
const resource_1 = require("./resource");
const utils_1 = require("./utils");
class PostsFilter {
    get query() {
        const query = {
            count: this.count,
            orderby: this.orderby,
            after: this.after ? this.after.toISOString() : null,
            filter: this.filter ? JSON.stringify(this.filter) : null,
            limit: this.limit,
            before: this.before ? this.before.toISOString() : null,
        };
        return utils_1.trimNullableValues(query);
    }
}
exports.PostsFilter = PostsFilter;
const basePath = '/api/2/posts';
class PostResourceImpl extends resource_1.Resource {
    async create(model) {
        const response = await this.request({
            path: basePath,
            method: resource_1.Methods.POST,
            body: model,
        });
        return model_1.toPost(response);
    }
    async list(filter) {
        const response = await this.request({
            path: basePath,
            method: resource_1.Methods.GET,
            query: filter === null || filter === void 0 ? void 0 : filter.query,
        });
        return response.map(model_1.toPost);
    }
}
exports.PostResourceImpl = PostResourceImpl;
