// beekeeper/dist/sdk/resource/profile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileResourceImpl = void 0;
const model_1 = require("../../model");
const user_1 = require("../../model/user");
const resource_1 = require("./resource");
const basePath = '/api/2/profiles';
class ProfileResourceImpl extends resource_1.Resource {
    async get(identifier) {
        const response = await this.request({
            path: `${basePath}/${identifier.toString()}`,
            method: resource_1.Methods.GET,
            query: {
                include_activities: 'false',
                include_totals: 'false',
            },
        });
        return model_1.toProfileContext(response);
    }
    /**
     * This endpoint is currently admin-only. In the future, we will provide a
     * method to access profiles by a normal user.
     */
    async list(filter) {
        const request = {
            path: '/api/2/users',
            method: resource_1.Methods.GET,
        };
        if (filter && filter.limit) {
            request.query = { limit: `${filter.limit}` };
        }
        const response = await this.request(request);
        return response.map(user_1.toUserFull);
    }
}
exports.ProfileResourceImpl = ProfileResourceImpl;
