// beekeeper/dist/sdk/resource/shift.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShiftResourceImpl = exports.ShiftsFilter = void 0;
const model_1 = require("../../model");
const resource_1 = require("./resource");
const utils_1 = require("./utils");
class ShiftsFilter {
    get query() {
        const query = {
            start: this.start ? this.start.toISOString() : null,
            end: this.end ? this.end.toISOString() : null,
        };
        return utils_1.trimNullableValues(query);
    }
}
exports.ShiftsFilter = ShiftsFilter;
class ShiftResourceImpl extends resource_1.Resource {
    async list(filter) {
        const response = await this.request({
            path: '/api/2/shifts',
            method: resource_1.Methods.GET,
            query: filter === null || filter === void 0 ? void 0 : filter.query,
        });
        if (!response || !response.length) {
            return [];
        }
        return response.map(model_1.toShift);
    }
}
exports.ShiftResourceImpl = ShiftResourceImpl;
